src/main/java/stirling/software/SPDF/config/AppConfig.java

```diff
@@ -135,4 +138,40 @@ public Predicate<Path> processOnlyFiles() {
             }
         };
     }
+
+    //    @Bean
+    //    public MemoryConfig mConfig() {
+    //        YAMLMapper yamlMapper = new YAMLMapper();
+    //        MemoryConfig memoryconfig = new MemoryConfig();
+    //        try (var input = Files.newInputStream(Paths.get("settings.yml"))) {
+    //            SimpleYamlImplementation yaml = new SimpleYamlImplementation();
+    //            // Load YAML content into MemoryConfig
+    //            memoryconfig = yamlMapper.readValue(input, MemoryConfig.class);
+    //        } catch (IOException e) {
+    //            // Handle the error and potentially set default values
+    //            logger.error("Error loading memory settings from YAML file", e);
+    //            // Optionally initialize default values here if necessary
+    //        }
+    //        return memoryconfig;
+    //    }
+
+    @Bean(name = "memoryConfig")
```

Comment:
This isnt right, you dont need to do any YAML mapping or anything...

https://github.com/Stirling-Tools/Stirling-PDF/blob/main/src/main/java/stirling/software/SPDF/model/ApplicationProperties.java
has
``@PropertySource(value = "file:./configs/settings.yml", factory = YamlPropertySourceFactory.class)``
Which means it will autoload these properties when autowired
this is the process used throughout AppConfig.java such as ``applicationProperties.getUi().getHomeDescription()`` etc

Adding the entry in settings.yml.template and then in ApplicationProperties should be all that's needed

Reply:
Yes, I updated it. Thank you. I tested the extracting images feature locally after changing the code and was successful, but I am not aware of why docker-compose tests are failing.